### PR TITLE
Add browser close friendly err (#687)

### PR DESCRIPTION
**Pull request by inancgumus**
This PR does the following:
* Allows executing `Browser.Close` only once.
* Warns users with "a friendly message" when it's called multiple times.

Notes:
* `Close()` is thread-safe as only the same VU or testBrowser use it.

You can try the result with this script:

```js
import { chromium } from 'k6/x/browser';

export default async function () {
    const browser = chromium.launch();
    const page = browser.newPage();

    await page
        .goto('https://test.k6.io/', { waitUntil: 'networkidle' })
        .finally(() => {
            page.close();
            browser.close();
        });

    await page
        .goto('https://test.k6.io/contacts.php', { waitUntil: 'networkidle' })
        .finally(() => {
            page.close();
            browser.close();
        });
}
```

It should print:

```
WARN[0002] Please call browser.close only once, and do not use the browser after calling close.
  category="Browser:Close" elapsed="0 ms" goroutine=84
```

### CODE REVIEW

File: common/browser.go [hunk 1]
```diff
@@ -62,6 +62,9 @@ type Browser struct {
 	sessionIDtoTargetIDMu sync.RWMutex
 	sessionIDtoTargetID   map[target.SessionID]target.ID
 
+	// Used to display a warning when the browser is reclosed.
+	closed bool
```
Comment (inancgumus) [inline common/browser.go:66]: I would use `atomic.Bool` and then I changed my mind as a single VU enters here.

File: common/browser.go [hunk 2]
```diff
@@ -391,6 +394,15 @@ func (b *Browser) newPageInContext(id cdp.BrowserContextID) (*Page, error) {
 
 // Close shuts down the browser.
 func (b *Browser) Close() {
+	if b.closed {
+		b.logger.Warnf(
+			"Browser:Close",
+			"Please call browser.close only once, and do not use continue using the browser after calling close.",
```
Comment (ka3de) [inline common/browser.go:400]: There is a typo in this message right?
I guess it should be:
```
Please call browser.close only once, and do not use the browser after calling close.
```
Maybe even omitting the last part of `calling close` or changing it for `after it`? I'm not native English speaker, so just suggesting :smile:

Comment (inancgumus) [reply common/browser.go]: @ka3de Rephrased like a native speaker here: https://github.com/grafana/xk6-browser/pull/803/commits/250b1e79838c9d636b5e8061ef35b180b1809ca9 😅

**Review by ka3de — APPROVED** [review]
Thanks for working on these optimizations!
LGTM.

**Review by ankur22 — APPROVED** [review]
LGTM

**Pull request merged**